src/utils/deleteDeployment.ts

```diff
@@ -26,30 +39,61 @@ export default async function deleteDeployment(
     activationURL: window.env.ACTIVATION_SERVICE_URL,
     relayURL: window.env.RELAY_DOMAIN,
   });
+  const hasGw = solutionWithGW.includes(type);
+  let defaultNameSpace = false;
 
   // remove deployment in namespace `type`
-  console.log({ grid });
   await grid.connect();
-  const obj = await _deleteDeployments(grid, name, configs, type, key);
+  const obj = await _deleteDeployments(grid, name, configs, type, key, defaultNameSpace, hasGw);
 
-  console.log({ obj });
-  if (obj["deleted"].length) {
+  if (obj.length) {
     return obj;
   }
 
   // remove deployments (vm, gw) in default namespace ``
-  grid.clientOptions.projectName = "";
-  await grid._connect();
-  console.log({ grid });
-  return await _deleteDeployments(grid, name, configs, "", key);
+  defaultNameSpace = true;
+  return await _deleteDeployments(grid, name, configs, type, key, defaultNameSpace, hasGw);
 }
 
-async function _deleteDeployments(grid, name, configs, type, key) {
+async function _deleteDeployments(grid, name, configs, type, key, defaultNameSpace = false, hasGw) {
   const domainName = await getUniqueDomainName(configs, name, type);
+
+  if (defaultNameSpace) {
+    grid.clientOptions.projectName = "";
+    await grid._connect();
+  }
+
   if (type === "Qvm") {
     const qvm = await grid.machines.getObj(name);
     await grid.qsfs_zdbs.delete({ name: qvm[0].mounts[0].name });
   }
-  await grid.gateway.delete_name({ name: domainName });
-  return await grid[key].delete({ name });
+
+  const deleted = await grid[key].delete({ name });
+  if (!hasGw || defaultNameSpace) return deleted;
+
+  const deletedgw = await _deleteDeploymentsGw(grid, domainName);
+  if (!deletedgw) throw new Error("Can't delete gateway contracts with name " + domainName);
+
+  return deleted;
+}
+
+async function _deleteDeploymentsGw(grid: GridClient, domainName: string) {
+  let deleted = await deleteGw(grid, grid.clientOptions.projectName, domainName);
+  if (deleted) return deleted;
+
+  deleted = await deleteGw(grid, "GatewayName", domainName);
+  if (deleted) return deleted;
+
+  deleted = await deleteGw(grid, "", domainName);
+  if (deleted) return deleted;
+
+  return null;
+}
```

Comment:
What is that!!

Reply:
there are three scenarios :

1. the gw is in the same namespace as the vm so in this case the deleteGw will be called once and return,
2. the gw is in `GatewayName` namespace in this case called twice and return 
3. the gw is in default namespace `""` so this case the deleteGw will be called three times.

case 2,3 are edge case for only some old deployments as there was a conflict in the gw namespace